deployment/cdk/opensearch-service-migration/lib/stack-composer.ts

```diff
@@ -228,49 +229,54 @@ export class StackComposer {
-            stage: stage,
-            defaultDeployId: defaultDeployId,
-            addOnMigrationDeployId: addOnMigrationDeployId,
-            ...props,
-        });
-
-        if (networkStack) {
-            openSearchStack.addDependency(networkStack)
-        }
-        this.stacks.push(openSearchStack)
+        // There is an assumption here that for any deployment we will always have a target cluster, whether that be a
+        // created cluster like below or an imported one
+        let openSearchStack
+        if (!targetClusterEndpoint) {
+            openSearchStack = new OpenSearchDomainStack(scope, `openSearchDomainStack-${deployId}`, {
+                version: version,
+                domainName: domainName,
+                dataNodeInstanceType: dataNodeType,
+                dataNodes: dataNodeCount,
+                dedicatedManagerNodeType: dedicatedManagerNodeType,
+                dedicatedManagerNodeCount: dedicatedManagerNodeCount,
+                warmInstanceType: warmNodeType,
+                warmNodes: warmNodeCount,
+                accessPolicies: accessPolicies,
+                useUnsignedBasicAuth: useUnsignedBasicAuth,
+                fineGrainedManagerUserARN: fineGrainedManagerUserARN,
+                fineGrainedManagerUserName: fineGrainedManagerUserName,
+                fineGrainedManagerUserSecretManagerKeyARN: fineGrainedManagerUserSecretManagerKeyARN,
+                enableDemoAdmin: enableDemoAdmin,
+                enforceHTTPS: enforceHTTPS,
+                tlsSecurityPolicy: tlsSecurityPolicy,
+                ebsEnabled: ebsEnabled,
+                ebsIops: ebsIops,
+                ebsVolumeSize: ebsVolumeSize,
+                ebsVolumeType: ebsVolumeType,
+                encryptionAtRestEnabled: encryptionAtRestEnabled,
+                encryptionAtRestKmsKeyARN: encryptionAtRestKmsKeyARN,
+                appLogEnabled: loggingAppLogEnabled,
+                appLogGroup: loggingAppLogGroupARN,
+                nodeToNodeEncryptionEnabled: noneToNodeEncryptionEnabled,
+                vpc: networkStack ? networkStack.vpc : undefined,
+                vpcSubnetIds: vpcSubnetIds,
+                vpcSecurityGroupIds: vpcSecurityGroupIds,
+                availabilityZoneCount: availabilityZoneCount,
+                domainRemovalPolicy: domainRemovalPolicy,
+                stackName: `OSMigrations-${stage}-${region}-${deployId}-OpenSearchDomain`,
+                description: "This stack contains resources to create/manage an OpenSearch Service domain",
+                stage: stage,
+                defaultDeployId: defaultDeployId,
+                addOnMigrationDeployId: addOnMigrationDeployId,
+                ...props,
+            });
+
+            if (networkStack) {
+                openSearchStack.addDependency(networkStack)
+            }
+            this.stacks.push(openSearchStack)
```

Comment:
what is the value of openSearchStack when targetClusterEndpoing was specified, will there be a side effect on this.stacks?